base/genericmemory.jl

```diff
@@ -207,12 +207,13 @@ getindex(A::Memory, c::Colon) = copy(A)
 
 ## Indexing: setindex! ##
 
-function setindex!(A::Memory{T}, x, i1::Int) where {T}
+function setindex!(A::GenericMemory{T}, x, i1::Int) where {T}
```

Comment:
I believe that this was intentionally not implemented (because for almost all algorithms you need a combination of reads and cas rather than reads and writes.